modules/mbedtls/Kconfig

```diff
@@ -151,6 +151,13 @@ config MBEDTLS_HEAP_SIZE
 	  be needed. For some dedicated and specific usage of mbedtls API, the
 	  1000 bytes might be ok.
 
+config MBEDTLS_HEAP_SELF_INIT
+	bool "Initialize the heap automatically at boot"
+	default y
+	help
+	  By default the heap will be initialized at Zephyr init. This option
+	  defers the heap initialization until explicitly called.
```

Comment:
Last things first, your suggested wording is great, thanks.
As to why it is needed... It is catering for a memory map where mbedTLS is separated out into its own area of rom. This area of rom exists in emulated environments but not in real hardware. We wanted to use the existing zephyr heap implementation for mbedTLS, just be in control of when it is called. This is perhaps a slightly unusual setup, but we figured it may well be useful for other users hence the PR. If the feeling is that this is too specialised a use case then no worries.

Reply:
thanks for the explanation, it's always helpful to understand the reason for a given patch.

I will let it be up to the maintainers to decide if this is valid additional ability in addition to existing use-cases.